File: Greenployee.MODELS/Model/OrdemServicoItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Greenployee.MODELS.Model
{
    public class OrdemServicoItem : BaseClass
    {

        [StringLength(100)]
        public string nmProduto { get; set; } = string.Empty;

        public decimal vlUnitario { get; set; }

        public int nrQuantidade { get; set; }

        public int idOrdemServico { get; set; }

        [ForeignKey("idOrdemServico")]
        public virtual OrdemServico OrdemServico { get; set; }
    }
}

File: GreenployeeClient/src/app/model/ordemServicoItem.ts
import { BaseClass } from "./baseClass";

export class OrdemServicoItem extends BaseClass {

    nmProduto!: string;
    vlUnitario!: number;
    nrQuantidade!: number;
    vlTotal!: number;
    idOrdemServico!: number;

    constructor() {
      super();
      this.nmProduto = "";
      this.vlUnitario = 0;
    }
}
